libraries/provider_firewall_windows.rb

```diff
@@ -68,11 +68,29 @@ def whyrun_supported?
       windows_file.content build_rule_file(new_resource.rules['windows'])
       windows_file.run_action(:create)
 
-      # if the file was changed, restart iptables
-      if windows_file.updated_by_last_action?
+      # if the file was changed
+      # OR
+      # if current_rule_count is not equal to the number of rules in file
```

Comment:
This logic seems very prone to breaking. Do we need to test that each of these cases works? Also, what if the current count happens to be the same as the defaults count, but we don't have the defaults? It feels very un-idempotent.